pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2487,3 +2487,369 @@ def test_mergeerror_on_left_index_mismatched_dtypes():
     df_2 = DataFrame(data=["X"], columns=["C"], index=[999])
     with pytest.raises(MergeError, match="Can only pass argument"):
         merge(df_1, df_2, on=["C"], left_index=True)
+
+
+class Test_AntiJoin:
```

Comment:
Could you put df1 and df2 into the __init__ method of this class and write down the tests explicitly? These parametrizations are really hard to read. Then you can write down one test for every case without duplicating unnecessary things.

And please call them left and right. Will look more closely after this

Reply:
I have seperated the tests into another file.. also split the tests for clarity..
Let me know how this looks.

Reply:
I will have a think on how to improve these tests to make them more compact and readable, but I wonder if issue #43096 might need to be addressed / discussed first, since I think it is relevant for this.